src/components/interaction-designer/block/BlockToolbar.vue

```diff
@@ -0,0 +1,177 @@
+        <div v-if="isDeleting">
+          <button
+            class="btn btn-light btn-xs"
+            @click.prevent="isDeleting = false">
+            <small>{{ trans('flow-builder.cancel') }}</small>
+          </button>
+          <button
+            class="btn btn-danger btn-xs ml-1"
+            @click.prevent="handleDeleteBlock">
+            <small>{{ trans('flow-builder.delete-block') }}</small>
+          </button>
+        </div>
+        <font-awesome-icon
+          v-if="!isDeleting"
+          v-b-tooltip.hover="trans('flow-builder.tooltip-delete-block')"
+          :icon="['far', 'trash-alt']"
+          class="fa-btn text-danger"
+          @click.prevent="isDeleting = true" />
+      </div>
+      <!--Duplicate-->
+      <div class="mr-1 ml-2">
+        <font-awesome-icon
+          v-if="isEditable"
+          v-b-tooltip.hover="trans('flow-builder.tooltip-duplicate-block')"
+          :icon="['fac', 'copy']"
+          class="fa-btn"
+          @click.prevent="handleDuplicateBlock" />
+      </div>
+      <!--Expand block editor-->
+      <div class="mr-1 ml-2">
+        <font-awesome-icon
+          v-if="isEditable"
+          v-b-tooltip.hover="trans('flow-builder.toggle-block-editor-tooltip')"
+          :icon="isEditorVisible ? ['fac', 'minimize'] : ['fac', 'expand']"
+          class="fa-btn"
+          @click.prevent="handleExpandMinimizeBlockEditor" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+
+import Component, {mixins} from 'vue-class-component'
+import Lang from '@/lib/filters/lang'
+import {Prop} from 'vue-property-decorator'
+import {IBlock} from '@floip/flow-runner'
+import {namespace} from 'vuex-class'
+
+const builderVuexNamespace = namespace('builder')
+const flowVuexNamespace = namespace('flow')
+const validationVuexNamespace = namespace('validation')
+
+@Component({})
+class BlockToolbar extends mixins(Lang) {
+  @Prop() readonly block!: IBlock
+  @Prop() readonly isBlockSelected!: boolean
+  @Prop() readonly isEditorVisible!: boolean
+
+  isDeleting = false
+
+  handleDeleteBlock() {
+    this.block_deselect({blockId: this.block.uuid})
+    this.flow_removeBlock({blockId: this.block.uuid})
+    this.removeValidationStatusesFor({key: `block/${this.block.uuid}`})
+    this.isDeleting = false
+    this.$emit('delete')
+  }
+
+  handleDuplicateBlock() {
+    this.flow_duplicateBlock({blockId: this.block.uuid}).then((duplicatedBlock) => {
+      this.$router.replace({
+        name: 'block-selected-details',
+        params: {blockId: duplicatedBlock.uuid},
+      })
+    })
+    this.$emit('duplicate')
+  }
+
+  handleExpandMinimizeBlockEditor() {
+    this.setIsBlockEditorOpen(!this.isEditorVisible)
+    let routerName = ''
+    if (this.isEditorVisible) {
+      routerName = 'block-selected-details'
+      this.$emit('minimize')
+    } else {
+      routerName = 'block-selected'
+      this.$emit('expand')
+    }
+
+    this.$router.history.replace({
```

Comment:
@poojahpatil90 , do you have an idea why this line fails in test ? https://app.travis-ci.com/github/FLOIP/flow-builder/builds/234046062#L1992
```
ERROR in /home/travis/build/FLOIP/flow-builder/src/components/interaction-designer/block/BlockToolbar.vue(113,18):
113:18 Property 'history' does not exist on type 'VueRouter'.
    111 |     }
    112 | 
  > 113 |     this.$router.history.replace({
        |                  ^
    114 |       name: routerName,
    115 |       params: {blockId: this.block.uuid},
    116 |     })
```